src/charm.py

```diff
@@ -1,88 +1,161 @@
+        self._kratos_url = None
+        self._port = None
+        self._reconfigure()
+        """New vars should come here"""
+
         self.service_patcher = KubernetesServicePatch(
-            self, [("api", 3000)]
+            self, [("Identity-Platform-Login-UI", self._port)]
         )
-        self.ingress = IngressPerAppRequirer(
+        self.public_ingress = IngressPerAppRequirer(
             self,
-            relation_name="ingress",
-            port=3000,
+            relation_name="public-ingress",
+            port=self._port,
             strip_prefix=True,
         )
 
-        self.framework.observe(self.on.kratos_ui_pebble_ready, self._on_kratos_ui_pebble_ready)
+        self.framework.observe(self.on.login_ui_pebble_ready, self._on_login_ui_pebble_ready)
         self.framework.observe(self.on.config_changed, self._on_config_changed)
-        self.framework.observe(self.ingress.on.ready, self._on_ingress_ready)
-
-    @property
-    def _container(self):
-        return self.unit.get_container(self._container_name)
+        self.framework.observe(self.public_ingress.on.ready, self._on_public_ingress_ready)
+        self.framework.observe(self.public_ingress.on.revoked, self._on_ingress_revoked)
 
-    def _on_kratos_ui_pebble_ready(self, event):
+    def _on_login_ui_pebble_ready(self, event: WorkloadEvent) -> None:
         """Define and start a workload using the Pebble API.
         """
-        self._container.add_layer("kratos_ui", self._pebble_layer, combine=True)
-        self._container.replan()
-        self.unit.status = ActiveStatus()
+        self._handle_status_update_config(event)
 
-    def _on_config_changed(self, event):
+    def _on_config_changed(self, event: ConfigChangedEvent) -> None:
         """Handle changed configuration.
         """
-        # Fetch the new config value
-        log_level = self.model.config["log-level"].lower()
+        self._handle_status_update_config(event)
+
+    def _reconfigure(self) -> None:
+        self._hydra_url = self.config.get('hydra_url', '')
+        logger.info("New value of hydra_url %s", self._hydra_url)
+        self._kratos_url = self.config.get('kratos_url', '')
+        logger.info("New value of kratos_url %s", self._kratos_url)
+        self._port = self.config.get('port', '')
+        logger.info("New value of port %s", self._port)
+
+    def _handle_status_update_config(self, event: HookEvent) -> None:
+        if not self._container.can_connect():
+            event.defer()
+            logger.info("Cannot connect to Login_UI container. Deferring the event.")
+            self.unit.status = WaitingStatus("Waiting to connect to Login_UI container")
+            return
+
+        self.unit.status = MaintenanceStatus("Configuration in progress")
+
+        self._reconfigure()
+        current_layer = self._container.get_plan()
+        new_layer = self._login_ui_layer
+        if current_layer.services != new_layer.services:
```

Comment:
I think that's a valid point. We use this check in other charms where the layer doesn't depend on config (it is only set once, on pebble ready event, and we don't expect it to be changed later). In such case we compare the layers first to avoid adding the same layer and replanning the container every time a similar method is run on multiple events.
(btw: `replan` won't restart the container if the layer has not changed. So even in above case, this check is not really needed, it's there just for optimization)

In your case, as long as the layer is subject to change due to config, this check is probably not needed.